### PR TITLE
nixos/ollama: split cuda and rocm from service test

### PR DESCRIPTION
**Pull request by abysssol**
## Description of changes

Cuda and rocm are platform specific, only working on linux. Also, ofborg won't run any of the service test at all because cuda is unfreely licensed. So, the rocm and cuda nodes have been moved into separate tests.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by abysssol** [comment]
@ofborg test ollama ollama-rocm

**Review by drupol — APPROVED** [review]

File: nixos/tests/all-tests.nix [hunk 1]
```diff
@@ -686,6 +686,8 @@ in {
   oddjobd = handleTestOn [ "x86_64-linux" "aarch64-linux" ] ./oddjobd.nix {};
   oh-my-zsh = handleTest ./oh-my-zsh.nix {};
   ollama = handleTest ./ollama.nix {};
+  ollama-cuda = handleTestOn ["x86_64-linux" "aarch64-linux"] ./ollama-cuda.nix {};
+  ollama-rocm = handleTestOn ["x86_64-linux" "aarch64-linux"] ./ollama-rocm.nix {};
```
Comment (drupol) [inline nixos/tests/all-tests.nix:690]: question (optional): Would it be possible to use `runTest` instead ?

Comment (abysssol) [reply nixos/tests/all-tests.nix:690]: What's the difference between `handleTest` and `runTest`? Does `runTest` use a vm too? Is `runTest` effective for testing nixos modules? Is there any documentation you could point me to (I couldn't find any)?

Comment (drupol) [reply nixos/tests/all-tests.nix:690]: `handleTest` and `runTest` are using the same technologies for testing. The only difference between the two is that writing tests using `runTest` seems to be simpler to write, a simple lambda function and it's done (there are examples in `nixos/tests/all-tests.nix`, check them out), while `handleTest` is a bit more complex with a call to `import ./make-test-python.nix`.

**Comment by abysssol** [comment]
@ofborg test ollama ollama-rocm

**Review by drupol — APPROVED** [review]

**Pull request merged**